overlord/state/taskrunner.go

```diff
@@ -255,12 +256,12 @@ func (r *TaskRunner) run(t *Task) {
 				t.SetStatus(UndoStatus)
 				r.state.EnsureBefore(0)
 			case UndoingStatus:
-				t.SetStatus(UndoneStatus)
+				taskStatusChanged = t.SetStatus(UndoneStatus)
 				fallthrough
 			case UndoneStatus:
 				next = t.WaitTasks()
 			}
-			if len(next) > 0 {
+			if len(next) > 0 || taskStatusChanged {
```

Comment:
the simplest change that I think makes sense would be to do:

chg := t.Change()
if len(next) > 0 ||  (chg != nil && chg.IsReady()) {
   ...
}

if this change doesn't work then I'm confused because that's the same condition at the top of TaskRunner.clean and we need to understand more of what happens. When is the relevant change becoming ready?

Reply:
This will probably work. I'll try now.